### PR TITLE
feat: add key transactions

### PR DESCRIPTION
**Pull request by sanderblue**
Resolves: #20

### CODE REVIEW

File: pkg/apm/key_transactions.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	if params.IDs != nil && len(params.IDs) > 0 {
+		paramsMap["filter[ids]"] = intArrayToString(params.IDs)
+	}
+
+	return paramsMap
+}
+
+// Converts an array of integers to a comma-separated list string.
+// Example: [1, 2, 3] will be converted to "1,2,3"
+func intArrayToString(integers []int) string {
```
Comment (sanderblue) [inline pkg/apm/key_transactions.go]: TODO: potentially put this in a centralized utils package or in a package-scoped `utils.go` file.

Comment (sanderblue) [reply pkg/apm/key_transactions.go]: Updated.

File: internal/utils/utils.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+// e.g [1, 2, 3] will be converted to "1,2,3".
+func IntArrayToString(integers []int) string {
+	sArray := []string{}
+
+	for _, n := range integers {
+		sArray = append(sArray, strconv.Itoa(n))
+	}
+
+	return strings.Join(sArray, ",")
+}
```
Comment (jthurman42) [inline internal/utils/utils.go]: Looping through append could be optimized since we know the length of `integers`:

```
func IntArrayToString(integers []int) string {
	sArray := make([]string, len(integers))

	for i, n := range integers {
		sArray[i] = strconv.Itoa(n)
	}

	return strings.Join(sArray, ",")
}
```

File: internal/utils/utils_test.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+)
+
+func TestIntArrayToString(t *testing.T) {
+	t.Parallel()
+
+	arr := []int{1, 2, 3, 4}
+	result := IntArrayToString(arr)
+
+	assert.Equal(t, "1,2,3,4", result)
+}
```
Comment (jthurman42) [inline internal/utils/utils_test.go]: Should add tests for Empty/Single/Multiple value slices:

```
func TestIntArrayToString(t *testing.T) {
	t.Parallel()

	var result string

	// empty
	result = IntArrayToString([]int{})
	assert.Equal(t, "", result)

	// single
	result = IntArrayToString([]int{1})
	assert.Equal(t, "1", result)

	// multiple
	result = IntArrayToString([]int{1, 2, 3, 4})
	assert.Equal(t, "1,2,3,4", result)
}
```

File: internal/utils/utils.go [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package utils
```
Comment (jthurman42) [inline internal/utils/utils.go]: nit: Utils is a great package name.  Should this file be called `strings` or something else more specific to the type of util it is?

Comment (sanderblue) [reply internal/utils/utils.go]: Fair question. My thought was that this was to not overthink that part since it's only one util function for now. And maybe as things are added, the proper "homes" would present themselves. That being said, in this case, I would think the file name for this util function would be `arrays.go` since the input is an array/slice.

File: internal/utils/utils_test.go (hunk 3 above)
Comment (sanderblue) [reply internal/utils/utils_test.go]: Updated, thanks!

File: internal/utils/utils.go (hunk 2 above)
Comment (sanderblue) [reply internal/utils/utils.go]: Updated, thanks!
